tests/unit/changeset-test.js

```diff
@@ -830,4 +838,166 @@ test('content can be an empty hash', function(assert) {
   let dummyChangeset = new Changeset(emptyObject, dummyValidator);
 
   assert.equal(dummyChangeset.toString(), 'changeset:[object Object]');
-})
+});
+
+test('it works with nested keys', function(assert) {
+  let expectedResult = {
+    org: {
+      asia: { sg: 'sg' },
+      usa: {
+        ca: 'ca',
+        ny: 'ny',
+        ma: { name: 'Massachusetts' }
+      }
+    }
+  };
+  set(dummyModel, 'org', {
+    asia: { sg: null },
+    usa: {
+      ca: null,
+      ny: null,
+      ma: { name: null }
+    }
+  });
+  let dummyChangeset = new Changeset(dummyModel, dummyValidator);
```

Comment:
Let's rename `dummyChangeset` to `c` for brevity. :) It'll cut down on some of the `dummyChangeset` noise.